### PR TITLE
Update RecordLogMessage() API to take getter function for context data

### PR DESCRIPTION
**Pull request by nr-ahemsath**
## Description

Implements https://issues.newrelic.com/browse/NEWRELIC-4758.  Includes unit tests to test context data filtration based on include and exclude lists in config (caveat: wildcard filtering support not implemented yet, will be implemented in a follow-on PR).

# Author Checklist
- [X] Unit tests, Integration tests, and Unbounded tests completed

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)
- [ ] Review Changelog

### CODE REVIEW

**Comment by nr-ahemsath** [comment]
Converted to draft since I forgot to add wildcard support in the include/exclude lists.

File: tests/Agent/UnitTests/Core.UnitTest/Wrapper/AgentWrapperApi/AgentWrapperApiTests.cs [hunk 1]
```diff
@@ -1675,18 +1714,83 @@ public void RecordLogMessage_WithTransaction_WithException_ThatHasHadLogsHarvest
...
+            Assert.AreEqual(priority, logEvent.Priority);
+        }
+
+        [TestCase(true, "", "", "key1,key2")]
+        [TestCase(true, "key1", "", "key1")]
+        [TestCase(true, "", "key1", "key2")]
+        [TestCase(true, "", "key1,key2", "")]
+        [TestCase(true, "key3", "", "")]
+        [TestCase(false, "", "", "")]
+        public void RecordLogMessage_ContextDataConfiguration(bool contextDataEnabled, string includeList, string excludeList, string expectedAttributeNames)
```
Comment (vuqtran88) [inline tests/Agent/UnitTests/Core.UnitTest/Wrapper/AgentWrapperApi/AgentWrapperApiTests.cs:1770]: Suggested additional test cases:

[TestCase(true, "key1,key2", "key2", "key1")]
[TestCase(false, "key1,key2", "key2", "")]

**Review by vuqtran88 — COMMENTED** [review]
I just gave this a first round of review with a question and a suggestion.

File: src/Agent/NewRelic/Agent/Core/Agent.cs [hunk 2]
```diff
@@ -552,6 +553,28 @@ internal void Detach(bool removeAsync, bool removePrimary)
             _transactionService.RemoveOutstandingInternalTransactions(removeAsync, removePrimary);
         }
 
+        private Dictionary<string, object> filterLogContextData(Dictionary<string, object> unfilteredContextData)
```
[the remark this answers is not in the document]
Comment (nr-ahemsath) [reply src/Agent/NewRelic/Agent/Core/Agent.cs:556]: No, I just forgot about our convention.  I'll fix it.

File: tests/Agent/UnitTests/Core.UnitTest/Wrapper/AgentWrapperApi/AgentWrapperApiTests.cs (hunk 1 above)
Comment (nr-ahemsath) [reply tests/Agent/UnitTests/Core.UnitTest/Wrapper/AgentWrapperApi/AgentWrapperApiTests.cs:1770]: Good idea.

**Review by vuqtran88 — APPROVED** [review]

**Review by chynesNR — APPROVED** [review]

**Pull request merged**